h/templates/groups/share.html.jinja2

```diff
@@ -32,17 +32,27 @@
         </ul>
     </div>
     {% endif %}
+    {% if member_usernames %}
+    <div class="group-members-list">
+      <ul class="group-members-list__list">
+        <p class="group-members-list__heading">Group Members:</p>
+        {% for member_username in member_usernames %}
+          <li>
+            <a href="{{ request.route_url('stream') }}?q=group:{{ group.pubid }} username: {{ member_username }}"
```

Comment:
The issue here is that the username you want to display is `chdorner` for example, but the username you want so search for is `chdorner@hypothes.is`.

You can call [this](https://github.com/hypothesis/h/blob/master/h/util/user.py#L24-L32) function for turning a username into a user id.